### PR TITLE
Update scopes and roles in the UI

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://github.com/ethyca/fides/issues/2743

This will impact https://github.com/ethyca/fides/pull/2739, I think only in when it lists out what roles are available

### Code Changes

* [x] Update TS autogenerated types
* [x] Update various home/tile restrictions based on the new more granular scopes
* [x] Use `total_scopes` instead of `scopes`

### Steps to Confirm

* [ ] things should work the same as before

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### Description Of Changes

_Write some things here about the changes and any potential caveats_

### CODE REVIEW

File: clients/admin-ui/cypress/e2e/routes.cy.ts [hunk 1]
```diff
@@ -30,13 +30,23 @@ describe("Routes", () => {
       cy.visit("/datastore-connection");
       cy.wait("@getConnectors");
       cy.getByTestId("connection-grid");
+      cy.visit("/privacy-requests/configure");
+      cy.getByTestId("privacy-requests-configure");
+    });
+
+    // This doesn't work right now due needing a fix for `exact` in the `nav-config` helpers
+    // This is the same issue as https://github.com/ethyca/fides/issues/2731
```
Comment (allisonking) [inline clients/admin-ui/cypress/e2e/routes.cy.ts:38]: this should be fixable, but I didn't have time to get to it

related to https://github.com/ethyca/fides/issues/2731

File: clients/admin-ui/cypress/fixtures/user-management/permissions.json [hunk 2]
```diff
@@ -71,9 +71,155 @@
     "validate:exec",
     "webhook:create_or_update",
     "webhook:delete",
+    "webhook:read",
```
Comment (pattisdr) [inline clients/admin-ui/cypress/fixtures/user-management/permissions.json:74]: Looks like this list is twice as long as it needs to be

File: clients/admin-ui/cypress/support/commands.ts [hunk 3]
```diff
@@ -82,9 +82,9 @@ declare global {
       login(): void;
       /**
        * Stub a user with the scopes associated with a role
-       * @example cy.assumeRole(RoleRegistry.ADMIN)
+       * @example cy.assumeRole(RoleRegistryEnum.ADMIN)
```
Comment (pattisdr) [inline clients/admin-ui/cypress/support/commands.ts:85]: new role is owner

File: clients/admin-ui/src/features/common/nav/v2/nav-config.test.ts [hunk 4]
```diff
@@ -56,8 +56,9 @@ describe("configureNavGroups", () => {
       config: NAV_CONFIG,
       hasSystems: true,
       userScopes: [
-        ScopeRegistry.CLI_OBJECTS_CREATE,
-        ScopeRegistry.CLI_OBJECTS_READ,
+        ScopeRegistryEnum.SYSTEM_CREATE,
```
Comment (pattisdr) [inline clients/admin-ui/src/features/common/nav/v2/nav-config.test.ts:59]: I think this should be reverted back to CLI_OBJECTS_CREATE - maybe DATASET_CREATE_OR_UPDATE can stay

File: clients/admin-ui/src/features/common/nav/v2/nav-config.test.ts [hunk 5]
```diff
@@ -83,10 +84,11 @@ describe("configureNavGroups", () => {
       hasSystems: true,
       hasPlus: true,
       userScopes: [
-        ScopeRegistry.DATAMAP_READ,
-        ScopeRegistry.CLI_OBJECTS_CREATE,
-        ScopeRegistry.CLI_OBJECTS_READ,
-        ScopeRegistry.CLI_OBJECTS_UPDATE,
+        ScopeRegistryEnum.DATAMAP_READ,
+        ScopeRegistryEnum.SYSTEM_CREATE,
```
Comment (pattisdr) [inline clients/admin-ui/src/features/common/nav/v2/nav-config.test.ts:88]: same here?

File: clients/admin-ui/src/features/common/nav/v2/nav-config.test.ts [hunk 6]
```diff
@@ -162,9 +164,10 @@ describe("configureNavGroups", () => {
         config: NAV_CONFIG,
         hasSystems: true,
         userScopes: [
-          ScopeRegistry.DATAMAP_READ,
-          ScopeRegistry.CLI_OBJECTS_CREATE,
-          ScopeRegistry.CLI_OBJECTS_READ,
+          ScopeRegistryEnum.DATAMAP_READ,
+          ScopeRegistryEnum.SYSTEM_CREATE,
```
Comment (pattisdr) [inline clients/admin-ui/src/features/common/nav/v2/nav-config.test.ts:168]: maybe here

File: clients/admin-ui/src/features/common/nav/v2/nav-config.test.ts [hunk 7]
```diff
@@ -264,22 +269,22 @@ describe("findActiveNav", () => {
       {
         path: "/add-systems",
         expected: true,
-        userScopes: [ScopeRegistry.CLI_OBJECTS_CREATE],
+        userScopes: [ScopeRegistryEnum.SYSTEM_CREATE],
```
Comment (pattisdr) [inline clients/admin-ui/src/features/common/nav/v2/nav-config.test.ts:272]: here too

File: clients/admin-ui/src/features/common/nav/v2/nav-config.ts [hunk 8]
```diff
@@ -57,34 +57,31 @@ export const NAV_CONFIG: NavConfigGroup[] = [
...
         ],
       },
       {
         title: "Manage datasets",
         path: "/dataset",
-        scopes: [
-          ScopeRegistry.CLI_OBJECTS_CREATE,
-          ScopeRegistry.CLI_OBJECTS_UPDATE,
-        ],
+        scopes: [ScopeRegistryEnum.DATASET_CREATE_OR_UPDATE],
```
Comment (pattisdr) [inline clients/admin-ui/src/features/common/nav/v2/nav-config.ts:78]: I think the current scopes on the /dataset endpoints are CLI_OBJECTS_CREATE/UPDATE - the systems work will change so it has a scope of CTL_DATASET CREATE/UPDATE, but I don't think this one is right ^

File: clients/admin-ui/src/features/common/nav/v2/nav-config.test.ts [hunk 9]
```diff
@@ -56,8 +56,9 @@ describe("configureNavGroups", () => {
       config: NAV_CONFIG,
       hasSystems: true,
       userScopes: [
-        ScopeRegistry.CLI_OBJECTS_CREATE,
-        ScopeRegistry.CLI_OBJECTS_READ,
+        ScopeRegistryEnum.SYSTEM_CREATE,
+        ScopeRegistryEnum.DATASET_CREATE_OR_UPDATE,
```
Comment (pattisdr) [inline clients/admin-ui/src/features/common/nav/v2/nav-config.test.ts:60]: These are actually on the ops endpoints not the ctl endpoints -

File: clients/admin-ui/src/flags.json [hunk 10]
```diff
@@ -1,7 +1,7 @@
 {
   "privacyRequestsConfiguration": {
     "development": true,
-    "test": false,
+    "test": true,
```
[the remark this answers is not in the document]
Comment (TheAndrewJackson) [reply clients/admin-ui/src/flags.json:4]: I think it was. The `clients/admin-ui/src/pages/privacy-requests/configure.tsx` file added this `data-testid="privacy-requests-configure"` to one of the divs. That id makes it easier to find elements while testing.

**Review by TheAndrewJackson — APPROVED** [review]

File: clients/admin-ui/cypress/e2e/routes.cy.ts (hunk 1 above)
Comment (pattisdr) [reply clients/admin-ui/cypress/e2e/routes.cy.ts:38]: resolving to get this merged and we can come back and fix this later -

**Pull request merged**